src/frontend/stores/component-data/component-data-store.ts

```diff
@@ -97,6 +97,45 @@ export class ComponentDataStore extends AbstractStore {
   }
 
   /**
+   * Get the nodes from root of the tree to given node.
+   * @param  {Object} node
+   */
+  private getPathToNode(node: any): any[] {
+    // The ID shows the path from root to given node.
+    let splitId: number[] = node.id.split('.');
+    let nodes: any[] = [];
+
+    // First index is for the component root:
+    let rootIndex: number = splitId.shift();
+    let root = this._componentData[rootIndex];
+    nodes.push(root);
+
+    // Next traverse the tree by index from ID.
+    splitId.reduce(function (last: any, subId: number) {
```

Comment:
@sumitarora used reduce to get the previous node (with initial value of root) in order to traverse the tree, since each element in splitId array is an index of the children array of the previous node from root to selected node